alf/environments/wrappers.py

```diff
@@ -27,24 +27,43 @@
 class FrameStack(gym.Wrapper):
     """Stack previous `stack_size` frames, applied to Gym env."""
 
-    def __init__(self, env, stack_size=4, channel_order='channels_last'):
+    def __init__(
+            self,
+            env,
+            stack_size=4,
+            channel_order='channels_last',
+            field_to_stack=None,
+    ):
         """Create a FrameStack object.
 
         Args:
+            env (gym.Space): gym environment.
             stack_size (int):
             channel_order (str): one of `channels_last` or `channels_first`.
                 The ordering of the dimensions in the images.
                 `channels_last` corresponds to images with shape
                 `(height, width, channels)` (Atari's default) while
                 `channels_first` corresponds to images with shape
                 `(channels, height, width)`.
+            field_to_stack (str): optional name of the key to the
+                Dict env.observation_space.  If specified, only use the
+                space corresponding to the key in FrameStack.
         """
         super().__init__(env)
         self._frames = collections.deque(maxlen=stack_size)
         self._channel_order = channel_order
         self._stack_size = stack_size
+        self._field_to_stack = field_to_stack
+
+        raw_space = self.env.observation_space
+        if field_to_stack:
+            assert hasattr(raw_space, 'spaces'), 'observation is not dict'
+            assert raw_space.spaces[field_to_stack], (
```

Comment:
If field_to_stack is not in raw_space, I think this will raise exception.
You can use `raw_space.contains(field_to_stack)`